src/commands.cpp

```diff
@@ -5856,42 +5835,40 @@ CommandQueryRecoveryLink::CommandQueryRecoveryLink(MegaClient *client, const cha
     tag = client->reqtag;
 }
 
-void CommandQueryRecoveryLink::procresult()
+bool CommandQueryRecoveryLink::procresult(Result r)
 {
     // [<code>,"<email>","<ip_address>",<timestamp>,"<user_handle>",["<email>"]]
 
+    //todo: haven't we already entered the array
```

Comment:
Thanks for trying that, you're right about the extra enterarray()/leavearray().  I think that was broken before, as the old Request code also already entered the array.  With the verifypassword result, the server is returning an object (quite a long one too), where I had assumed from the old code that it would just be an error or OK, as that code followed the normal pattern, however now I see it varies slightly, reporting API_OK in the other branch after storeobject().  I've fixed that up, and it is the only case like that.   thanks!

Reply:
Confirmed with a successful email change via megacli.